File: Data Structures & Algorithms/permutation-string/submission-1.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        for i in range(len(s2)):
            for j in range(i, len(s2)):
                if "".join(sorted(s1)) == "".join(sorted(s2[i:j+1])):
                    return True

        return False

File: Data Structures & Algorithms/anagram-groups/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let res = {};

        for (let i of strs) {
            let count = new Array(26).fill(0);

            for (let c of i) {
                count[c.charCodeAt(0) - "a".charCodeAt(0)] += 1;
            }

            const key = count.join(",");
            if (!res[key]) {
                res[key] = [];
            }
            res[key].push(i)
        }

        return Object.values(res);
    }
}
